### PR TITLE
restore tweaks

### PR DESCRIPTION
**Pull request by czue**
possibly easiest read commit by commit.

biggest change is the addition of a new simplified sync log format (not yet used but will be soon)

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/phone/dbaccessors/sync_logs_by_user.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    return SyncLog.view(
+        "phone/sync_logs_by_user",
+        startkey=[user_id, {}],
+        endkey=[user_id],
+        descending=True,
+        limit=1,
+        reduce=False,
+        include_docs=True,
+    ).one()
+
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/dbaccessors/sync_logs_by_user.py]: W391 blank line at end of file

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -78,18 +78,26 @@ def __init__(self, case_id, *args, **kwargs):
...
+    duration = IntegerProperty()        # in seconds
+
+    # owner_ids_on_phone stores the ids the phone thinks it's the owner of.
+    # This typically includes the user id,
+    # as well as all groups that that user is a member of.
+    owner_ids_on_phone = StringListProperty()
+
+
+
+class SyncLog(AbstractSyncLog):
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E303 too many blank lines (3)

File: corehq/ex-submodules/casexml/apps/phone/data_providers/case/utils.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ def _get_required_updates(self):
         pairing. Should be a list of actions like [create, update, close]
         """
         ret = []
-        if not self.sync_token or not self.sync_token.phone_has_case(self.case.get_id):
+        if not self.sync_token or not self.sync_token.phone_is_holding_case(self.case.get_id):
```
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/data_providers/case/utils.py]: I think this is a bugfix. Should have been true even if the phone just had the dependent case (which it wouldn't have been with the previous code call)

**Comment by czue** [comment]
(don't merge until tests pass, obviously)

File: corehq/ex-submodules/casexml/apps/case/util.py [hunk 4]
```diff
@@ -5,7 +5,7 @@
 from casexml.apps.case import const
 from casexml.apps.case.dbaccessors import get_indexed_case_ids
 from casexml.apps.case.sharedmodels import CommCareCaseIndex
-from casexml.apps.phone.models import SyncLogAssertionError, SyncLog
+from casexml.apps.phone.models import SyncLogAssertionError, SyncLog, get_properly_wrapped_sync_log
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/util.py]: F401 'SyncLog' imported but unused

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 5]
```diff
@@ -128,33 +152,43 @@ def invalidate_cached_payloads(self):
...
-                logging.warn("assertion failed: %s" % msg)
-                self.has_assert_errors = True
+    # anything prefixed with 'tests_only' is only used in tests
+    def tests_only_get_cases_on_phone(self):
+        raise NotImplementedError()
+
+    def test_only_clear_cases_on_phone(self):
+        raise NotImplementedError()
+
+class SyncLog(AbstractSyncLog):
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E302 expected 2 blank lines, found 1

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 6]
```diff
@@ -312,7 +312,8 @@ class RestoreState(object):
     reasons.
     """
     def __init__(self, domain, user, params):
```
Comment (emord) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: Should you also change the param domain to project? Seems more clear that way

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: almost definitely. will do

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 7]
```diff
@@ -411,15 +412,15 @@ class RestoreConfig(object):
     """
 
     def __init__(self, domain=None, user=None, params=None, cache_settings=None):
```
Comment (emord) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: Same thing here

File: corehq/ex-submodules/casexml/apps/phone/restore.py (hunk 6 above)
Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: (done)

**Comment by czue** [comment]
:white_check_mark: @snopoke @emord if this is g2g would be nice to get it in. if you're still looking then I'll just keep pushing changes to this branch

**Comment by snopoke** [comment]
:white_check_mark: from me

**Pull request merged**